File: backend/IMIS/IMIS.Domain/PgsDeliverableAccomplishment .cs
using System.ComponentModel.DataAnnotations.Schema;
using Base.Primitives;

namespace IMIS.Domain
{
    public class PgsDeliverableAccomplishment : Entity<long>
    {
        public required long PgsDeliverableId { get; set; }

        [ForeignKey(nameof(PgsDeliverableId))]
        public virtual PgsDeliverable? PgsDeliverable { get; set; }

        public DateTime PostingDate { get; set; }

        public required string UserId { get; set; }

        [ForeignKey(nameof(UserId))]
        public virtual User? User { get; set; }

        public required double PercentAccomplished { get; set; }
        public string? Remarks { get; set; }
     
        public string? AttachmentPath { get; set; }
    }
}

File: frontend/imis/lib/performance_governance_system/deliverable_status_monitoring/models/pgs_deliverable_accomplishment.dart
import 'package:imis/utils/date_time_converter.dart';
import 'package:json_annotation/json_annotation.dart';
part 'pgs_deliverable_accomplishment.g.dart';

@JsonSerializable()
class PgsDeliverableAccomplishment {
  int id;
  int? pgsDeliverableId;

  @JsonKey()
  @DateTimeConverter()
  DateTime postingDate;
  String? userId;
  int? status;
  DateTime byWhen;
  int? percentAccomplished;
  String? remarks;
  String? attachmentPath;
  String? deliverableName;
  String? pgsStatus;
  final bool? isDeleted;
  final String? rowVersion;
  String? auditorRemarks;
  PgsDeliverableAccomplishment(
    this.id,
    this.pgsDeliverableId,
    this.postingDate,
    this.userId,
    this.status,
    this.byWhen,
    this.percentAccomplished,
    this.remarks,
    this.attachmentPath,
    this.deliverableName,
    this.pgsStatus,
    this.auditorRemarks, {
    this.isDeleted,
    this.rowVersion,
  });

  factory PgsDeliverableAccomplishment.fromJson(Map<String, dynamic> json) =>
      _$PgsDeliverableAccomplishmentFromJson(json);

  Map<String, dynamic> toJson() => _$PgsDeliverableAccomplishmentToJson(this);
}
